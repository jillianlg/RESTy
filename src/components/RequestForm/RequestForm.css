  .RequestForm{
    margin-top: .3vw;
    margin-right: .3vw;
    width: 81.9vw;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    font-family: 'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande', 'Lucida Sans', Arial, sans-serif;

    div{
      display: flex;
      width: 100%;
      justify-content: flex-start;
      align-items: center;
      flex-direction: row;
    
      input[type="radio"]{
        visibility: hidden;
        height: 0;
        width: 0;
        margin-left: 0;
      }

      label{
        padding: 4px 16px;
        font-size: 12px;
        cursor: pointer;
      }

      input[type="radio"]:checked+label{
        background-color: lightgray;
        color: slategray;
        align-items: center;
      }
    }
  
  input{
    margin-bottom: 8px; 
  }

  button{
    margin-left: 2vw;
    width: 50%;
    align-self: center;
    font-size: 12px;
    font-family: 'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande', 'Lucida Sans', Arial, sans-serif;
  }

  textarea{
    margin-top: 5px;
    width: 75%;
    height: 6vw;
    align-self: center;
    font-family: 'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande', 'Lucida Sans', Arial, sans-serif;
    font-size: 10px;
  }
}

.url{
  width: 100%;
  height: 25px;
  font-size: 12px;
  margin-top: 5px;
  font-family: 'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande', 'Lucida Sans', Arial, sans-serif;
}