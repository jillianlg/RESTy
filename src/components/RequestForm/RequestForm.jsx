import React from 'react';

const RequestForm = ({ url, onSubmit, onChange }) => {
  <div>
    <form onSubmit={onSubmit}>
      <input 
        type="text" 
        name="url" 
        placeholder="URL" 
        value={url} 
        onChange={onChange} />
    </form>
  </div>;
};

export default RequestForm;
