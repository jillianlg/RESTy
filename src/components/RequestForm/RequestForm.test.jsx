import React from 'react';
import { render, cleanup } from '@testing-library/react';
import RequestForm from '../RequestForm/RequestForm';


describe('RequestForm component', () => {
  afterEach(() => cleanup());
  it('renders request in the form', () => {
    const { asFragment } = render(
      <RequestForm
        url="http://"
        method="GET"
        body="{}"
        onSubmit={jest.fn()}
        onChange={jest.fn()}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
