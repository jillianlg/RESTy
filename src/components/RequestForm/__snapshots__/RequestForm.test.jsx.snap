// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RequestForm component renders request in the form 1`] = `
<DocumentFragment>
  <div>
    <form
      class="RequestForm"
    >
      <input
        class="url"
        name="url"
        placeholder="URL"
        type="text"
        value="http://"
      />
      <section>
        <div>
          <input
            checked=""
            id="get"
            name="method"
            type="radio"
            value="GET"
          />
          <label
            for="get"
          >
            GET
          </label>
          <input
            id="post"
            name="method"
            type="radio"
            value="POST"
          />
          <label
            for="post"
          >
            POST
          </label>
          <input
            id="put"
            name="method"
            type="radio"
            value="PUT"
          />
          <label
            for="put"
          >
            PUT
          </label>
          <input
            id="patch"
            name="method"
            type="radio"
            value="PATCH"
          />
          <label
            for="patch"
          >
            PATCH
          </label>
          <input
            id="delete"
            name="method"
            type="radio"
            value="DELETE"
          />
          <label
            for="delete"
          >
            DELETE
          </label>
        </div>
      </section>
      <button>
        Submit
      </button>
      <textarea
        name="body"
        placeholder="Raw JSON Body"
      >
        {}
      </textarea>
    </form>
  </div>
</DocumentFragment>
`;
